File: app/src/main/java/com/a531tracker/HomeScreen.java
package com.a531tracker;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.a531tracker.Database.DatabaseHelper;

public class HomeScreen extends AppCompatActivity {
    static final int SET_TRAINING_MAX_CODE = 1;
    static final int UPDATE_TRAINING_MAX_CODE = 2;

    public static String[] compoundLifts = new String[]{"Bench", "Overhand Press", "Squat", "Deadlift"};

    private DatabaseHelper db;
    private Context mContext;
    private Integer cycleValue;

    private Button benchView, deadliftView, pressView, squatView, cycleView;
    private Button homeButton, settingsButton;

    private TextView[] newTvArray;
    private TextView cycleDisplay;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        db = new DatabaseHelper(this);
        mContext = this;

        cycleDisplay = findViewById(R.id.cycleDisplay);

        setViews();
        setNav();
        setListeners();
    }


    protected void onStart(){
        super.onStart();
        startCycle();
        setCurrentTrainingValues();
    }


    private void setCurrentTrainingValues() {
        for(int i=0; i < compoundLifts.length; i++){
            newTvArray[i].setText(
                    String.valueOf(
                            db.getLifts(compoundLifts[i]).getTraining_max()
                    )
            );
        }
    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if(requestCode == SET_TRAINING_MAX_CODE && resultCode == RESULT_OK){
            //resetLifValuesArray();
        } else if (requestCode == UPDATE_TRAINING_MAX_CODE && resultCode == RESULT_OK){
            //resetLifValuesArray();
        }
        super.onActivityResult(requestCode, resultCode, data);
    }


    public void openCompoundWeek(String compound, String swap){
        Intent intent = new Intent(getApplicationContext(), Week.class);
        intent.putExtra("Compound", compound);
        intent.putExtra("Swap", swap);
        intent.putExtra("Cycle", cycleValue);
        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
        startActivity(intent);
    }


    public void alertBuilder(String title, String message, String extraMessage, boolean cancelable, final boolean noLiftsFound){
        AlertDialog.Builder dialog = new AlertDialog.Builder(this, R.style.AlertDialogTheme);
        dialog.setTitle(title)
                .setMessage(message)
                .setCancelable(cancelable)
                .setPositiveButton(R.string.ok_text, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        if(noLiftsFound) {
                            Intent intent = new Intent(mContext, SetMaxes.class);
                            intent.putExtra("NewLifts", true);
                            startActivityForResult(intent, SET_TRAINING_MAX_CODE);
                        }
                    }
                });
        if(!noLiftsFound){
            TextView tv = new TextView(this);
            tv.setGravity(Gravity.CENTER);
            tv.setTextColor(ContextCompat.getColor(mContext, R.color.colorOrange));
            tv.setText(extraMessage);
            tv.setTextSize(getResources().getDimension(R.dimen.text_8));
            dialog.setView(tv);
        } else {
            dialog.setMessage(message);
        }
        AlertDialog alertDialog = dialog.create();
        alertDialog.show();
    }


    private void startCycle(){
        try{
            if(db.startCycle()){
                cycleValue = db.getCycle();
            }
        } catch (Exception e){
            cycleValue = 1;
            e.printStackTrace();
        }
        String cycleText = "You are currently on Cycle #" + cycleValue;
        cycleDisplay.setText(cycleText);
    }


    private void setViews(){
        benchView = findViewById(R.id.bench_btn);
        deadliftView = findViewById(R.id.deadlift_btn);
        pressView = findViewById(R.id.press_btn);
        squatView = findViewById(R.id.squat_btn);
        cycleView = findViewById(R.id.update_cycle_btn);

        homeButton = findViewById(R.id.home_button);
        settingsButton = findViewById(R.id.settings_button);

        newTvArray = new TextView[]{findViewById(R.id.bench_value), findViewById(R.id.press_value), findViewById(R.id.squat_value), findViewById(R.id.deadlift_value)};
    }


    private void setNav(){
        navHome();
        navSettings();
    }


    //Buttons

    public void setListeners(){
        benchButton();
        deadliftButton();
        pressButton();
        squatButton();
        updateCycleButton();
    }


    private void benchButton() {
        benchView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openCompoundWeek("Bench", "Overhand Press");
            }
        });
    }


    private void deadliftButton(){
        deadliftView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openCompoundWeek("Deadlift", "Squat");
            }
        });
    }


    private void pressButton(){
        pressView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openCompoundWeek("Overhand Press", "Bench");
            }
        });
    }


    private void squatButton(){
        squatView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openCompoundWeek("Squat", "Deadlift");
            }
        });
    }



    private void updateCycleButton(){
        cycleView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(mContext, UpdateValues.class);
                startActivityForResult(intent, UPDATE_TRAINING_MAX_CODE);
            }
        });
    }


    private void navHome(){
        homeButton.setEnabled(false);
    }


    private void navSettings(){
        settingsButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(mContext, Settings.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/a531tracker/HomeScreen.kt
package com.a531tracker

import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Gravity
import android.widget.Button
import android.widget.TextView

import com.a531tracker.database.DatabaseHelper

class HomeScreen : AppCompatActivity() {

    private var db: DatabaseHelper? = null
    private var mContext: Context? = null
    private var cycleValue: Int? = null

    private var benchView: Button? = null
    private var deadliftView: Button? = null
    private var pressView: Button? = null
    private var squatView: Button? = null
    private var cycleView: Button? = null
    private var homeButton: Button? = null
    private var settingsButton: Button? = null

    private var newTvArray: Array<TextView>? = null
    private var cycleDisplay: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        db = DatabaseHelper(this)
        mContext = this

        cycleDisplay = findViewById(R.id.cycleDisplay)

        setViews()
        setNav()
        setListeners()
    }


    override fun onStart() {
        super.onStart()
        startCycle()
        setCurrentTrainingValues()
    }


    private fun setCurrentTrainingValues() {
        for (i in compoundLifts.indices) {
            newTvArray!![i].text = db!!.getLifts(compoundLifts[i]).training_max.toString()
        }
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == SET_TRAINING_MAX_CODE && resultCode == Activity.RESULT_OK) {
            //resetLifValuesArray();
        } else if (requestCode == UPDATE_TRAINING_MAX_CODE && resultCode == Activity.RESULT_OK) {
            //resetLifValuesArray();
        }
        super.onActivityResult(requestCode, resultCode, data)
    }


    fun openCompoundWeek(compound: String, swap: String) {
        val intent = Intent(applicationContext, Week::class.java)
        intent.putExtra("Compound", compound)
        intent.putExtra("Swap", swap)
        intent.putExtra("Cycle", cycleValue)
        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
        startActivity(intent)
    }


    fun alertBuilder(title: String, message: String, extraMessage: String, cancelable: Boolean, noLiftsFound: Boolean) {
        val dialog = AlertDialog.Builder(this, R.style.AlertDialogTheme)
        dialog.setTitle(title)
                .setMessage(message)
                .setCancelable(cancelable)
                .setPositiveButton(R.string.ok_text) { dialogInterface, i ->
                    if (noLiftsFound) {
                        val intent = Intent(mContext, SetMaxes::class.java)
                        intent.putExtra("NewLifts", true)
                        startActivityForResult(intent, SET_TRAINING_MAX_CODE)
                    }
                }
        if (!noLiftsFound) {
            val tv = TextView(this)
            tv.gravity = Gravity.CENTER
            tv.setTextColor(ContextCompat.getColor(mContext!!, R.color.colorOrange))
            tv.text = extraMessage
            tv.textSize = resources.getDimension(R.dimen.text_8)
            dialog.setView(tv)
        } else {
            dialog.setMessage(message)
        }
        val alertDialog = dialog.create()
        alertDialog.show()
    }


    private fun startCycle() {
        try {
            if (db!!.startCycle()) {
                cycleValue = db!!.cycle
            }
        } catch (e: Exception) {
            cycleValue = 1
            e.printStackTrace()
        }

        val cycleText = "You are currently on Cycle #" + cycleValue!!
        cycleDisplay!!.text = cycleText
    }


    private fun setViews() {
        benchView = findViewById(R.id.bench_btn)
        deadliftView = findViewById(R.id.deadlift_btn)
        pressView = findViewById(R.id.press_btn)
        squatView = findViewById(R.id.squat_btn)
        cycleView = findViewById(R.id.update_cycle_btn)

        homeButton = findViewById(R.id.home_button)
        settingsButton = findViewById(R.id.settings_button)

        newTvArray = arrayOf(findViewById(R.id.bench_value), findViewById(R.id.press_value), findViewById(R.id.squat_value), findViewById(R.id.deadlift_value))
    }


    private fun setNav() {
        navHome()
        navSettings()
    }


    //Buttons

    fun setListeners() {
        benchButton()
        deadliftButton()
        pressButton()
        squatButton()
        updateCycleButton()
    }


    private fun benchButton() {
        benchView!!.setOnClickListener { openCompoundWeek("Bench", "Overhand Press") }
    }


    private fun deadliftButton() {
        deadliftView!!.setOnClickListener { openCompoundWeek("Deadlift", "Squat") }
    }


    private fun pressButton() {
        pressView!!.setOnClickListener { openCompoundWeek("Overhand Press", "Bench") }
    }


    private fun squatButton() {
        squatView!!.setOnClickListener { openCompoundWeek("Squat", "Deadlift") }
    }


    private fun updateCycleButton() {
        cycleView!!.setOnClickListener {
            val intent = Intent(mContext, UpdateValues::class.java)
            startActivityForResult(intent, UPDATE_TRAINING_MAX_CODE)
        }
    }


    private fun navHome() {
        homeButton!!.isEnabled = false
    }


    private fun navSettings() {
        settingsButton!!.setOnClickListener {
            val intent = Intent(mContext, Settings::class.java)
            startActivity(intent)
        }
    }

    companion object {
        internal val SET_TRAINING_MAX_CODE = 1
        internal val UPDATE_TRAINING_MAX_CODE = 2

        var compoundLifts = arrayOf("Bench", "Overhand Press", "Squat", "Deadlift")
    }
}
